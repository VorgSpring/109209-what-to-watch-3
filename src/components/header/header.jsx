import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import {Link} from 'react-router-dom';
import {UserBlockContainer} from '../user-block/user-block.jsx';
import {RoutePaths} from '../../constants/route-paths';

export const Header = ({pageTitle, isMovieCard}) => (
  <Fragment>
    <h1 className='visually-hidden'>WTW</h1>

    <header className={`page-header ${isMovieCard ? `movie-card` : `user-page`}__head`}>
      <div className='logo'>
        <Link className='logo__link' to={RoutePaths.MAIN}>
          <span className='logo__letter logo__letter--1'>W</span>
          <span className='logo__letter logo__letter--2'>T</span>
          <span className='logo__letter logo__letter--3'>W</span>
        </Link>
      </div>

      {pageTitle && <h1 className='page-title user-page__title'>
        {pageTitle}
      </h1>}

      {isMovieCard && <UserBlockContainer />}
    </header>
  </Fragment>
);

Header.defaultProps = {
  pageTitle: null,
  isMovieCard: true
};

Header.propTypes = {
  pageTitle: PropTypes.string,
  isMovieCard: PropTypes.bool
};
