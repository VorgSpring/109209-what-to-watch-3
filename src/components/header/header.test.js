import React from 'react';
import {BrowserRouter} from 'react-router-dom';
import {Header} from './header.jsx';
import renderer from 'react-test-renderer';

jest.mock(`../user-block/user-block.jsx`, () => ({
  UserBlockContainer() {
    return <user-block />;
  }
}));

describe(`Header`, () => {
  it(`should renders correctly`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders correctly movie type`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header movieType />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders correctly user type`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header userType />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders correctly with show user block`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header showUserBlok />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders correctly with title`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header title='blah-blah' />
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`should renders correctly with children`, () => {
    const tree = renderer.create(
        <BrowserRouter>
          <Header>
            <div>blah-blah</div>
          </Header>
        </BrowserRouter>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
