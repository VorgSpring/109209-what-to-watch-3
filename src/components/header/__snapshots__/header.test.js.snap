// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should renders correctly 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header
        false
        false"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
  </header>,
]
`;

exports[`Header should renders correctly movie type 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header
        movie-card__head
        false"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
  </header>,
]
`;

exports[`Header should renders correctly user type 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header
        false
        user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
  </header>,
]
`;

exports[`Header should renders correctly with children 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header
        false
        false"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <div>
      blah-blah
    </div>
  </header>,
]
`;

exports[`Header should renders correctly with show user block 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header
        false
        false"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <user-block />
  </header>,
]
`;

exports[`Header should renders correctly with title 1`] = `
Array [
  <h1
    className="visually-hidden"
  >
    WTW
  </h1>,
  <header
    className="page-header
        false
        false"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      blah-blah
    </h1>
  </header>,
]
`;
